src/engine/utils.cpp

```diff
@@ -208,4 +208,116 @@ void Utils::prepare_inputs(const std::vector<Sequence*>& batch,
   input_params->token_ids_lens = torch::tensor(token_ids_lens_vec, torch::kInt);
 }
 
+void Utils::prepare_validate_inputs(const std::vector<Sequence*>& batch,
+                                    int32_t block_size,
+                                    torch::Tensor* flatten_token_ids,
+                                    torch::Tensor* flatten_positions,
+                                    torch::Tensor* seq_idxes,
+                                    InputParameters* input_params,
+                                    SamplingParameters* sampling_params) {
+  std::vector<int32_t> flatten_tokens_vec;
```

Comment:
Just a heads up, this part may need to change after attention kernel change.

Reply:
yes, we can change this part follow up with new attention kernels.